src/results.cpp

```diff
@@ -47,47 +47,51 @@ Results::Results(SharedRealm r, Table& table)
 {
 }
 
-Results::Results(SharedRealm, TableView, DescriptorOrdering) { throw std::runtime_error("not implemented"); }
-Results::Results(std::shared_ptr<Realm>, LstBase&, util::Optional<Query>, SortDescriptor)
-{ throw std::runtime_error("not implemented"); }
+Results::Results(std::shared_ptr<Realm> r, std::shared_ptr<LstBase> list)
+: m_realm(std::move(r))
+, m_list(list)
+, m_mode(Mode::List)
+{
+}
 
-#if 0
-Results::Results(SharedRealm r, LinkListRef lv, util::Optional<Query> q, SortDescriptor s)
+Results::Results(std::shared_ptr<Realm> r, TableView tv, DescriptorOrdering o)
+: m_realm(std::move(r))
+, m_table_view(std::move(tv))
+, m_descriptor_ordering(std::move(o))
+, m_mode(Mode::TableView)
+{
+    m_table = TableRef(&m_table_view.get_parent());
+}
+
+Results::Results(std::shared_ptr<Realm> r, std::shared_ptr<LnkLst> lv, util::Optional<Query> q, SortDescriptor s)
 : m_realm(std::move(r))
 , m_link_list(lv)
 , m_mode(Mode::LinkList)
 {
-    m_table = lv->get_target_table();
+    m_table = TableRef(&m_link_list->get_target_table());
     if (q) {
         m_query = std::move(*q);
         m_mode = Mode::Query;
     }
     m_descriptor_ordering.append_sort(std::move(s));
 }
 
-Results::Results(SharedRealm r, TableView tv, DescriptorOrdering o)
-: m_realm(std::move(r))
-, m_table_view(std::move(tv))
-, m_descriptor_ordering(std::move(o))
-, m_mode(Mode::TableView)
-{
-    m_table = m_table_view.get_parent();
-}
-#endif
-
 Results::Results(const Results&) = default;
 Results& Results::operator=(const Results&) = default;
-Results::Results(Results&& other) = default;
-Results& Results::operator=(Results&& other) = default;
+Results::Results(Results&&) = default;
+Results& Results::operator=(Results&&) = default;
 
 bool Results::is_valid() const
 {
     if (m_realm) {
         m_realm->verify_thread();
     }
 
-//    if (m_table && !m_table->is_valid())
-//        return false;
+    if (((const Table*)m_table) != nullptr)
```

Comment:
This is an awfully longwinded way to write `if (m_table)`

Reply:
Explanation added